src/ophyd_async/panda/panda.py

```diff
@@ -80,66 +50,31 @@ def _block_name_number(block_name: str) -> Tuple[str, Optional[int]]:
     return block_name, None
 
 
-def _remove_inconsistent_blocks(pvi_info: Dict[str, PVIEntry]) -> None:
+def _remove_inconsistent_blocks(pvi_info: Optional[Dict[str, PVIEntry]]) -> None:
     """Remove blocks from pvi information.
 
     This is needed because some pandas have 'pcap' and 'pcap1' blocks, which are
     inconsistent with the assumption that pandas should only have a 'pcap' block,
     for example.
 
     """
+    if pvi_info is None:
+        return
     pvi_keys = set(pvi_info.keys())
     for k in pvi_keys:
         kn = re.sub(r"\d*$", "", k)
         if kn and k != kn and kn in pvi_keys:
             del pvi_info[k]
 
 
-async def pvi(
-    pv: str, ctxt: Context, timeout: float = DEFAULT_TIMEOUT
-) -> Dict[str, PVIEntry]:
-    try:
-        result = await pvi_get(pv, ctxt, timeout=timeout)
-    except TimeoutError as exc:
-        raise NotConnected(pv) from exc
-
-    _remove_inconsistent_blocks(result)
-    return result
-
-
 class PandA(Device):
-    _ctxt: Optional[Context] = None
-
     pulse: DeviceVector[PulseBlock]
     seq: DeviceVector[SeqBlock]
     pcap: PcapBlock
 
-    def __init__(self, prefix: str, name: str = "") -> None:
+    def __init__(self, prefix: str, name: str) -> None:
```

Comment:
Still want to default to blank name

Reply:
From BlueAPI I think we'd prefer not to allow defaulting to blank name- especially if there's multiple pandas per beamline.

Reply:
But from creation via DeviceCollector then we would like this, so to ease this I suggest we put the name creation and checking in the dodal/blueapi layers, and have an option name in the ophyd layer

Reply:
ceded